File: Algorithms/896.Monotonic-Array/monotonic_array.py
# -*- coding: utf-8 -*-


class Solution:
    def isMonotonic(self, A: List[int]) -> bool:
        flag = None
        for i in range(len(A) - 1):
            diff = A[i] - A[i+1]
            if diff == 0:
                continue
            diff = diff < 0
            if flag == None:
                flag = diff
            if diff != flag:
                return False
        return True

File: Algorithms/896.Monotonic-Array/monotonic_array.go
func isMonotonic(A []int) bool {
	flag := 0
	for i := 0; i < len(A)-1; i++ {
		if A[i] == A[i+1] {
			continue
		}

		cur_flag := 0
		if A[i] < A[i+1] {
			cur_flag = 1
		} else {
			cur_flag = 2
		}

		if flag == 0 {
			flag = cur_flag
			continue
		}
		if flag != cur_flag {
			return false
		}
	}
	return true
}
